### PR TITLE
Add a sampling method to HIRM.

### PR DESCRIPTION
**Pull request by emilyfertig**
I added `incorporate_sample` to `Relation` instead of taking care of everything in HIRM in anticipation of it being useful for our eventual HIRM/PClean hybrid that will enforce a unique-valued index domain in samples as well.

### CODE REVIEW

**Comment by emilyfertig** [comment]
Fixes #125

File: cxx/hirm.hh [hunk 1]
```diff
@@ -65,6 +65,11 @@ class HIRM {
 
   double logp_score() const;
```
Comment (ThomasColthurst) [inline cxx/hirm.hh:67]: Please add brief descriptions of these methods.

File: cxx/relation.hh [hunk 2]
```diff
@@ -21,44 +21,58 @@ class Relation {
...
-  virtual double cluster_or_prior_logp(std::mt19937* prng, const T_items& items, const ValueType& value) const = 0;
+  virtual double cluster_or_prior_logp(std::mt19937* prng, const T_items& items,
+                                       const ValueType& value) const = 0;
 
-  virtual ValueType sample_at_items(std::mt19937* prng, const T_items& items) const = 0;
+  virtual ValueType sample_at_items(std::mt19937* prng,
+                                    const T_items& items) const = 0;
 
-  virtual void incorporate_to_cluster(const T_items& items, const ValueType& value) = 0;
+  virtual void incorporate_sample(std::mt19937* prng, const T_items& items) = 0;
```
Comment (ThomasColthurst) [inline cxx/relation.hh:40]: Please add a description of this new method.

**Review by ThomasColthurst — APPROVED** [review]

File: cxx/relation.hh (hunk 2 above)
Comment (emilyfertig) [reply cxx/relation.hh:41]: Done.

File: cxx/hirm.hh (hunk 1 above)
Comment (emilyfertig) [reply cxx/hirm.hh:67]: Done.

**Pull request merged**